packages/app/src/sections/dashboard/RewardsTab.tsx

```diff
@@ -0,0 +1,382 @@
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { claimMultipleAllRewards } from 'state/staking/actions'
+import { setSelectedEpoch } from 'state/staking/reducer'
+import {
+	selectEpochData,
+	selectKwentaRewards,
+	selectOpRewards,
+	selectSelectedEpoch,
+	selectSnxOpRewards,
+} from 'state/staking/selectors'
+import { selectNetwork, selectWallet } from 'state/wallet/selectors'
+import media from 'styles/media'
+
+type EpochValue = {
+	period: number
+	start: number
+	end: number
+	label: string
+}
+
+const RewardsTabs: FC<TradingRewardProps> = ({
+	period = 0,
+	start = 0,
+	end = Math.floor(Date.now() / 1000),
+}) => {
+	const { t } = useTranslation()
+	const dispatch = useAppDispatch()
+	const network = useAppSelector(selectNetwork)
+	const isL2 = useIsL2()
+	const epochData = useAppSelector(selectEpochData)
+	const selectedEpoch = useAppSelector(selectSelectedEpoch)
+	const walletAddress = useAppSelector(selectWallet)
+	const kwentaRewards = useAppSelector(selectKwentaRewards)
+	const opRewards = useAppSelector(selectOpRewards)
+	const snxOpRewards = useAppSelector(selectSnxOpRewards)
+
+	const futuresFeeQuery = useGetFuturesFeeForAccount(walletAddress!, start, end)
+	const futuresFeePaid = useMemo(() => {
+		const t: FuturesFeeForAccountProps[] = futuresFeeQuery.data ?? []
+		return t
+			.map((trade) => formatEther(trade.feesPaid.sub(trade.keeperFeesPaid).toString()))
+			.reduce((acc, curr) => acc.add(wei(curr)), ZERO_WEI)
+	}, [futuresFeeQuery.data])
+
+	const totalFuturesFeeQuery = useGetFuturesFee(start, end)
+	const totalFuturesFeePaid = useMemo(() => {
+		const t: FuturesFeeProps[] = totalFuturesFeeQuery.data ?? []
+		return t
+			.map((trade) => formatEther(trade.feesKwenta.toString()))
+			.reduce((acc, curr) => acc.add(wei(curr)), ZERO_WEI)
+	}, [totalFuturesFeeQuery.data])
+
+	const handleClaimAll = useCallback(() => {
+		dispatch(claimMultipleAllRewards())
+	}, [dispatch])
+
+	const estimatedKwentaRewardQuery = useGetFile(
+		`trading-rewards-snapshots/${network === 420 ? `goerli-` : ''}epoch-current.json`
+	)
+	const estimatedKwentaReward = useMemo(
+		() => BigNumber.from(estimatedKwentaRewardQuery?.data?.claims[walletAddress!]?.amount ?? 0),
+		[estimatedKwentaRewardQuery?.data?.claims, walletAddress]
+	)
+
+	const estimatedOpQuery = useGetFile(
+		`trading-rewards-snapshots/${network === 420 ? `goerli-` : ''}epoch-current-op.json`
+	)
+	const estimatedOp = useMemo(
+		() => BigNumber.from(estimatedOpQuery?.data?.claims[walletAddress!]?.amount ?? 0),
+		[estimatedOpQuery?.data?.claims, walletAddress]
+	)
+
+	const claimDisabledAll = useMemo(() => kwentaRewards.add(opRewards).add(snxOpRewards).lte(0), [
+		opRewards,
+		snxOpRewards,
+		kwentaRewards,
+	])
+
+	const ratio = useMemo(() => {
+		return futuresFeePaid.gt(0) && totalFuturesFeePaid.gt(0)
+			? futuresFeePaid.div(totalFuturesFeePaid)
+			: ZERO_WEI
+	}, [futuresFeePaid, totalFuturesFeePaid])
+
+	const REWARDS = [
```

Comment:
I think we can put this behind a useMemo instead as currently it will reformat this data on each render.

Reply:
Thanks. Updated